syntax = "proto3";

package transport_catalogue_serialize;

message RouterSettings {
    double bus_wait_time = 1;
	double bus_velocity = 2;
}

message Point {
    double x = 1;
    double y = 2;
}

message Rgb {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
}

message Rgba {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
    double opacity = 4;
}

message Color {
    string str = 1;
    Rgb rgb = 2;
    Rgba rgba = 3;
    int32 what_ = 4;
}

message RenderSettings {
    double width = 1;
    double height = 2;

    double padding = 3;

    double line_width = 4;

    double stop_radius = 5;

    int32 bus_label_font_size = 6;

    Point bus_label_offset = 7;

    int32 stop_label_font_size = 8;

    Point stop_label_offset = 9;

    Color underlayer_color = 10;
    
    double underlayer_width = 11;

    repeated Color color_palette = 12;
}

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message RoadDistance {
    string name = 1;
    int32 dist = 2;
}

message Stop {
    string name = 1;
	Coordinates coordinate = 2;
    repeated RoadDistance road_distances = 3;
}

message Bus {
    string name = 1;
    repeated string stops = 2;
	bool is_round_ = 3;
}

message TransportCatalogue {
    RouterSettings router_settings = 1;
    RenderSettings render_settings = 2;
    repeated Stop stops = 3;
    repeated Bus buses = 4;
}